kivy/lang/builder.py

```diff
@@ -396,8 +396,17 @@ def load_string(self, string, **kwargs):
                 self.files.append(fn)
 
             if parser.root:
-                widget = Factory.get(parser.root.name)()
-                self._apply_rule(widget, parser.root, parser.root)
+                widget = Factory.get(parser.root.name)(__no_builder=True)
+                rule_children = []
+                append_rule_child = rule_children.append
```

Comment:
The builder functions should accept a list, like in my example code, to which it appends the children and not a function that is called.

The former is a very clean pattern, as it's just appending to a list, and it may or may not be useful for other things besides our desired usage. The latter, is much more general as it accepts any function so it can work for other needs, but is also therefore more complex. Right now there's no reason to think accepting any function will be ever needed so there's no reason to have this extra complexity. So please change it accepting a list rather than an append function.